src/test/java/net/pms/util/FileUtilTest.java

```diff
@@ -120,6 +120,238 @@ public void testGetFileNameWithRewriting() throws Exception {
+			        	if(metadata_rel.has("month")) {
+			        		rel_date = rel_date + "-" + Integer.toString(metadata_rel.get("month").getAsInt());
+			        	}
+			        	if(metadata_rel.has("date")) {
+			        		rel_date = rel_date + "-" + Integer.toString(metadata_rel.get("date").getAsInt());
+			        	}
+		        	}
+	        		elem = metadata.get("title");
+	        		if( elem != null ) {
+	        			try {
+	        				assertThat(tvEpisodeName).isEqualTo(elem.getAsString());
+	        			}
+        				catch(NullPointerException ex) {
+        					if(todo) { logger.warn( "testGetFileNameMetadata/title would fail for TODO test " + original ); }
+        					else { throw(new AssertionError(ex)); }
+        				}
+        				catch(AssertionError err) {
+        					if(todo) { logger.warn( "testGetFileNameMetadata/title would fail for TODO test " + original ); }
+        					else { throw(err); }
+        				}
+	        		}
+	        	}
+	        	else if( "tv-series-episodes".equals(metadata.get("type").getAsString()) ) {
+	        		logger.debug( "Doing tv-series-episodes " + original );
+	        		// A single episode or an episode range, cannot have episode title or date
+	        		elem = metadata.get("series");
+	        		if( elem != null ) {
+	        			try {
+	        				assertThat(movieOrShowName).isEqualTo(elem.getAsString());
+	        			}
+        				catch(NullPointerException ex) {
+        					if(todo) { logger.warn( "testGetFileNameMetadata/series would fail for TODO test " + original ); }
+        					else { throw(new AssertionError(ex)); }
+        				}
+        				catch(AssertionError err) {
+        					if(todo) { logger.warn( "testGetFileNameMetadata/series would fail for TODO test " + original ); }
+        					else { throw(err); }
+        				}
+	        		}
+	        		elem = metadata.get("season");
+	        		if( elem != null ) {
+	        			assertThat(tvSeason).isEqualTo(elem.getAsInt()); 
+	        		}
+	        		elem = metadata.get("episodes");
+	        		if( elem != null ) {
+	        			String range = "";
+	        			for( JsonElement elem2: elem.getAsJsonArray() ) {
+	        				range = range + "-" + String.format("%02d",elem2.getAsInt());
+	        			}
+	        			try {
+	        				assertThat(tvEpisodeNumber).isEqualTo(range.substring(1));
+	        			}
+        				catch(AssertionError err) {
+        					if(todo) { logger.warn( "testGetFileNameMetadata/episodes would fail for TODO test " + original ); }
+        					else { throw(err); }
+        				}
+	        		}
+	        	}
+	        	else if( "movie".equals(metadata.get("type").getAsString()) ) {
+	        		logger.debug( "Doing movie " + original );
+	        		elem = metadata.get("title");
+	        		if( elem != null ) {
+	        			try  {
+	        				assertThat(movieOrShowName).isEqualTo(elem.getAsString());
+	        			}
+        				catch(NullPointerException ex) {
```

Comment:
There is quite a lot in this file that doesn't match our codestyle but I can fix that after we merge

Reply:
Is the codestyle explained somewhere? I can try to fix if you have a pointer.

Reply:
It's not something that we enforce because I really want to encourage contributions, I don't want to turn people away if they write a bit differently :) But if you are interested we basically follow the way Google recommends (or rather, they have started recommending it the way we already were doing it :)) https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style